File: longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0];
        
        for i in range(1,len(strs)):
            prefix = self.getPrefix(prefix,strs[i])
        return prefix
        
    def getPrefix(self,str1:str,str2:str)->str:
        i=j=0
        while i<len(str1) and j<len(str2):
            if str1[i]!=str2[j]:
                break
            i+=1
            j+=1
        return str1[:i]

File: longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let sstr = strs[0];
    
    for(let i=1;i<strs.length;i++){
        if(strs[i].length<sstr.length){
            sstr = strs[i]
        }
    }
    
    let prefix = "";
    for(let i=0;i<sstr.length;i++){
        prefix+=sstr[i];
        if(!strs.every(str=>str.startsWith(prefix))){
            prefix = prefix.slice(0,-1);
            break;
        }
    }
    
    return prefix;
};